### PR TITLE
Shell escape user editable input in automated git commands

### PR DESCRIPTION
**Pull request by jesseleite**

### CODE REVIEW

File: src/Git/Git.php [hunk 1]
```diff
@@ -252,4 +252,18 @@ protected function push($gitRoot)
...
+    /**
+     * Shell escape string for use in git commands.
+     *
+     * @param string $string
+     * @return string
+     */
+    protected function shellEscape(string $string)
+    {
+        $string = str_replace('"', '', $string);
+        $string = str_replace("'", '', $string);
```
Comment (nl-ix-arjan) [inline src/Git/Git.php:265]: This `shellEscape` function is now used to escape Git user names (line 243). It strips out single quotes. What happens when a user's name legitimately has single quotes in them, such as e.g. "A'idah"?

File: src/Git/Git.php [hunk 2]
```diff
@@ -240,8 +240,8 @@ protected function getCommandContext($paths, $message)
         return [
             'paths' => collect($paths)->implode(' '),
             'message' => $message,
```
Comment (nl-ix-arjan) [inline src/Git/Git.php:242]: Why not escape the message and the paths as well?

Comment (jasonvarga) [reply src/Git/Git.php:242]: Those aren't user input.

Comment (nl-ix-arjan) [reply src/Git/Git.php:242]: Ok, but if you have a file with a quote in its name, they may cause unintended interpretation by the shell before the `git` commands gets to see them?

Comment (nl-ix-arjan) [reply src/Git/Git.php:242]: Additionally, the commit messages appear to be generated by code and can be translated, such as e.g. here:

https://github.com/statamic/cms/blob/7911d611c737c4d2c3edc8bc9e03c4ab153d1a4d/src/Events/AssetDeleted.php#L16

Without escaping those messages here, everyone that writes those messages in the code or in the translations has to be aware that they will be included in shell commands and thus have to remember to always "pre-escape" them.

Comment (jasonvarga) [reply src/Git/Git.php:242]: Fair points, we can escape these too.

File: src/Git/Git.php (hunk 1 above)
Comment (jasonvarga) [reply src/Git/Git.php:265]: Valid. 👍

Comment (jasonvarga) [reply src/Git/Git.php:265]: We're leaving this fix as is for now. It's a bit naive as it'll strip out the quotes from cases like O'Brien like you said, but that's cosmetic - it'll only be in the git log.

File: src/Git/Git.php (hunk 2 above)
Comment (jasonvarga) [reply src/Git/Git.php]: Commit message is now escaped. Paths aren't, but they're only ever going to be what's defined in `config/statamic/git.php`. If people are adding paths in there that contain quotes, we can address that separately.

**Pull request merged**